### PR TITLE
fix serviceName when backend.service.port uses name instead of number

### PR DESCRIPTION
**Pull request by ianhe8x**
**What this PR does / why we need it**:
For IngressV1beta1 the servicename in kong is composed of `ingress.Namespace`.`Backend.ServiceName`.`Backend.ServicePort.String()`
if servicePort is represented as string name, it will use the string name. otherwise it will use the port number.

But for IngressV1, `Backend.Service.Port.Number` is used. It causes an issue when I have both http and ws in the ingress and they use different ports (defined by string name). Only one kong service will be created, the last part of service name is `".0" `

**Special notes for your reviewer**:
I'm not sure where is the most preferred place to put `func serviceBackendPortToStr` pls advice then I will make change.

### CODE REVIEW

File: pkg/parser/translate.go [hunk 1]
```diff
@@ -18,6 +18,13 @@ import (
 	knative "knative.dev/networking/pkg/apis/networking/v1alpha1"
 )
 
+func serviceBackendPortToStr(port networkingv1.ServiceBackendPort) string {
+	if port.Name != "" {
+		return port.Name
+	}
+	return strconv.Itoa(int(port.Number))
```
Comment (mflendrich) [inline pkg/parser/translate.go:25]:
```suggestion
	if port.Name != "" {
		return fmt.Sprintf("pname-%s", port.Name)
	}
	return fmt.Sprintf("pnum-%d", port.Number)
```

**Review by mflendrich — CHANGES_REQUESTED** [review]
Thank you for this contribution. The problem stated exists indeed, but the proposed solution can potentially generate non-unique names in a corner case where the port name is a string representing a number (e.g. `name: "80"`). 

Please make the added function a little more defensive by qualifying the generated name with a prefix depending on the source of generation. An example solution attached in a comment below. Also, please add a unit test. Otherwise the change looks fine :+1:.

**Comment by ianhe8x** [comment]
Hi, I don't know why integration tests failed. It timeouted on master when I run it locally.

**Comment by mflendrich** [comment]
Thanks @ianhe8x,

Please target `next` with your PR as this is a change that has too much impact to go into a patch release.

> I don't know why integration tests failed
The git checkout error looks like a GitHub Actions machinery/misconfiguration issue that we haven't seen on other PRs - please change PR base (this will probably require rebasing) and let's see if the error still persists.

**Comment by ianhe8x** [comment]
closes in favour of #1245

**Pull request closed without merging**